bundles/customer-registration/src/FondOfOryx/Zed/CustomerRegistration/Persistence/CustomerRegistrationEntityManager.php

```diff
@@ -77,6 +77,26 @@ public function flagCustomerAsGdprAccepted(CustomerTransfer $customerTransfer):
             ->setUpdatedAt($this->convertDate($customerEntity->getUpdatedAt()));
     }
 
+    /**
+     * @param \Generated\Shared\Transfer\CustomerTransfer $customerTransfer
+     *
+     * @return \Generated\Shared\Transfer\CustomerTransfer
+     */
+    public function persistRegistrationKeyToCustomer(CustomerTransfer $customerTransfer): CustomerTransfer
+    {
+        $customerTransfer->requireRegistrationKey();
+        $customerEntity = $this->resolveCustomerEntity($customerTransfer);
+
+        $customerEntity->setRegistrationKey($customerTransfer->getRegistrationKey());
+        $customerEntity->save();
+
+        return $customerTransfer
+            ->setRegistrationKey($customerTransfer->getRegistrationKey())
```

Comment:
What is that line for, if the customerTransfer already includes the registrationKey

Reply:
Its a fallback. Not every customer has the registration key in the db so there will be optionally generated a new one if one is missing. Can bei enabled/disabled by plugin